lib/exercises.rb

```diff
@@ -2,29 +2,98 @@
 
 # This method will return an array of arrays.
 # Each subarray will have strings which are anagrams of each other
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n), goes through the length of the string array
+# Space Complexity: O(n), size of the output hash depends on the size of the string array
 
 def grouped_anagrams(strings)
-  raise NotImplementedError, "Method hasn't been implemented yet!"
+  return [] if strings.empty?
+  
+  output_hash = {}
+  
+  strings.each do |word|
+    letters = {}
+    
+    word.each_char do |letter|
+      if letters[letter].nil?
+        letters[letter] = 1
+      else
+        letters[letter] +=1
+      end
+    end
+    
+    if !output_hash[letters].nil?
+      output_hash[letters] << word
+    else
+      output_hash[letters] = [word]
+    end
+  end
+  
+  return output_hash.values
+  
 end
 
 # This method will return the k most common elements
 # in the case of a tie it will select the first occuring element.
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n^2)
+# Space Complexity: O(n)
 def top_k_frequent_elements(list, k)
```

Comment:
I would say O(kn) since you have max and delete nested in the k.times loop. 

Otherwise this works